aten/src/ATen/Dispatch.h

```diff
@@ -20,6 +20,14 @@
     return __VA_ARGS__();                                               \
   }
 
+// BFLOAT16_DISPATCH() macro dispatches bfloat16 datatype
+// on only ROCm platform.
+#ifdef __HIP_PLATFORM_HCC__
+#define BFLOAT16_DISPATCH(...)  AT_PRIVATE_CASE_TYPE(at::ScalarType::BFloat16, at::BFloat16, __VA_ARGS__)
+#else
+#define BFLOAT16_DISPATCH(...)
```

Comment:
This name doesn't make it very obvious what is going on...

Reply:
If it's supposed to be HIP only, put that in the macro name

Reply:
Changed the macro name to `BFLOAT16_DISPATCH_ROCM()`. Hope that makes it clear.